### DISCUSSION TITLE
How to download a file in Remix?

### DISCUSSION THREAD

**Discussion by janhesters** [remix-run/remix#5533] [Q&A]
I want to allow my users to download a text file.

How would I go about this in Remix?

I tried exposing the `.txt` file via a resource route, but then the user navigates away from the page. But I'd the file to just download, or maybe have them pick a download location. In any case, I want the user to stay on the page.

**Comment by Jackardios** [comment]
You can send a `Content-Disposition: attachment` header from resource route loader:
```ts
// routes/some[.txt].ts
export function loader() {
  const content = 'some text content...';
  return new Response(content, {
    status: 200,
    headers: {
      'Content-Type': 'plain/text',
      'Content-Disposition': 'attachment; filename="some.txt"',
    },
  });
}
```
or add `download` attribute to Link tag
```tsx
<Link to="some.txt" reloadDocument download="some.txt">some.txt file</Link>
```

**Comment by Yu-Keu** [comment]
Jackardios' answer works!
Also, I'm a noob, and for those who are wondering how to do this with `<Form>` instead of a link and want to send data to the action, you just need to add `reloadDocument `to the form, like this:
```
<Form reloadDocument>
  <!-- form fields go here -->
</Form>
```
